Makes sures pkgconfig has the right version
Index: apriltag/CMakeLists.txt
===================================================================
--- apriltag.orig/CMakeLists.txt
+++ apriltag/CMakeLists.txt
@@ -19,7 +19,10 @@ else()
     target_link_libraries(${PROJECT_NAME} Threads::Threads m)
 endif()
 
-set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION 3 VERSION 3.0.0)
+set(APRILTAG_VERSION 3.0.0)
+set(APRILTAG_ABI_VERSION 3)
+
+set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION ${APRILTAG_ABI_VERSION} VERSION ${APRILTAG_VERSION})
 
 include(GNUInstallDirs)
 target_include_directories(${PROJECT_NAME} PUBLIC
@@ -53,6 +56,7 @@ install(EXPORT apriltag
 
 FILE(READ apriltag.pc.in PKGC)
 STRING(REGEX REPLACE "^prefix=" "prefix=${CMAKE_INSTALL_PREFIX}" PKGC_CONF "${PKGC}" )
+STRING(REGEX REPLACE "@APRILTAG_VERSION@" "${APRILTAG_VERSION}" PKGC_CONF "${PKGC}" )
 FILE(WRITE ${PROJECT_BINARY_DIR}/apriltag.pc ${PKGC_CONF})
 install(FILES "${PROJECT_BINARY_DIR}/apriltag.pc" DESTINATION "lib/pkgconfig/")
 
@@ -70,4 +74,3 @@ endif(OpenCV_FOUND)
 
 # install example programs
 install(TARGETS apriltag_demo RUNTIME DESTINATION bin)
-
Index: apriltag/apriltag.pc.in
===================================================================
--- apriltag.orig/apriltag.pc.in
+++ apriltag/apriltag.pc.in
@@ -5,6 +5,6 @@ libdir=${exec_prefix}/lib
 
 Name: apriltag
 Description: AprilTag detector library
-Version: 0.10.0
+Version: @APRILTAG_VERSION@
 Cflags: -I${includedir}
 Libs: -L${libdir} -lapriltag
